manifests/init.pp

```diff
@@ -0,0 +1,48 @@
+# @summary Profile for installing and managing Java.
+#
+# Profile for installing and managing Java.
+#
+# @param install_lock_files
+#   Boolean - Install lock files or not.
+#
+# @param lock_files
+#   Hash of lock files (and parent dir(s)).
+#
+# @param pkg_list
+#   Hash of package names to install
+#   Keys must be an OS Family name
+#   Values must be a Hash of the format:
+#       Key = package name
+#           A package name can be prefixed with "-" to remove it from the list.
+#           In this way, a higher level hiera layer can remove a package added
+#           by a lower, common layer.
+#
+#       Value (optional):
+#           1. <EMPTY> in which case, the package will be installed
+#           2. <HASH> conataining valid attributes for the 
+#              "package" native puppet resource type
+#
+# @example
+#   include profile_java
+class profile_java (
+  Boolean $install_lock_files,
+  Hash    $lock_files,
+  Hash    $pkg_list,
```

Comment:
This parameter can/should just be changed to an array. I cannot envision a scenario where it would ever need to be a hash. The main reason I suggest this change is just to keep the usage similar to other existing NCSA shared profiles.

Reply:
Yeah, probably not.

With profile_additional_packages we do sometimes take advantage of having the packages themselves be hashes, so we can add extra params for Yum, e.g., to disable certain repos for that transaction. But if the variable itself is an array I think it still may work for the packages to be hashes.

And if not, I guess I am also having trouble envisioning a specific/likely scenario _for Java_ where we'd need that. My (slightly updated) thought _in a cluster environment_ is that I only see us installing 1+ Java versions via Red Hat RPMs and that anything from another source would get installed in the shared filesystem.

And if we need to update things later about how this variable is structured, it's easy enough.

So, yep, changed this to an array and updated the Hiera data and doc string/REFERENCE.md too.